src/deployconfig.jl

```diff
@@ -55,6 +55,23 @@ function documenter_key_previews(cfg::DeployConfig)
     return get(ENV, "DOCUMENTER_KEY_PREVIEWS", documenter_key(cfg))
 end
 
+function _decode_key_content(keycontent)
+    # If `keycontent` contains both `-----BEGIN` AND `-----END`, then we assume that it is
+    # a plaintext private key, and we don't try to Base64-decode it.
+    #
+    # Otherwise, we conclude that it must be a Base64-encoded private key, and we try to
+    # Base64-decode it.
+    keycontent_stripped = strip(keycontent)
+    is_plaintext_key = occursin("-----BEGIN", keycontent_stripped) && occursin("-----END", keycontent_stripped)
+    if is_plaintext_key
+        @debug "This looks like a plaintext private key, so we won't try to Base64-decode it"
+        return keycontent_stripped * '\n'
```

Comment:
```suggestion
    is_plaintext_key = occursin("-----BEGIN", keycontent) && occursin("-----END", keycontent)
    if is_plaintext_key
        @debug "This looks like a plaintext private key, so we won't try to Base64-decode it"
        return keycontent * '\n'
```

Reply:
What is the reason for appending the trailing newline? Is it automatically stripped when pasting the key into a GH secret but necessary for Git?

Reply:
Yeah, IIUC, the private key needs to have a newline at the end (e.g. https://askubuntu.com/questions/1344482/ssh-private-key-file-requires-n-at-the-end-of-file-or-cause-load-key-id-rsa).